### ISSUE TITLE
Pipe lost typing for inference generic 

### ISSUE THREAD

**Issue by RomanRolginSky** [ReactiveX/rxjs#6362]
## Bug Report

**Current Behavior**
Pipe operator resolves type to `never` instead of correct type of Observable

**Expected behavior**
Operators' types should be resolved correctly.
The correct type inferred from generic correctly

**Reproduction**
- REPL or Repo link:
https://codesandbox.io/s/ts-rx-generics-types-issue-forked-olcxu?file=/src/index.ts

**Environment**
- Runtime: Reproduces locally on Mac OS 11.1, Node v.12.16.1
- RxJS version: 6.6.7 in codesandbox.io, 6.1.3 locally

**Additional context/Screenshots**
I tried to reproduce this behavior without rxjs but without success so seems like its somehow relates to rxjs pipe

**Comment by cartant** [comment]
I don't believe that this is an RxJS problem. Rather, I think it is more likely to be a TypeScript issue.

> I tried to reproduce this behavior without rxjs but without success

I would suggest you try reproducing the problem in the TypeScript Playground and reporting it as a TypeScript problem if/when you are able to do so. You can use [this Playground](https://www.typescriptlang.org/play?#code/C4TwDgpgBAKgFgSwHYHMA8MB8UC8UDeAUFFAG4CGANgK4QBcsA3MVGApGgIIA0UAQrwDCmABQsSAezCcGAeTAZenTN3FQpfOQp78VaqYK1oBUYSwCUDeMnTDmAX2ahIUeWgCSvWdjwjgiVCsA9HdMc1xsa1Q0b2ZCABMIAGNKcgAnaAAzaiQk4AQJJCgAW3IFGF4AJVEwNIkAK2TgBhEKGnpYcJxsSstXcqrMZkSU9KycvIKi4DKMUSQIAA9mqFaqWisu7FIJBHi+twrYIcIE5NSMqCTCgGdgKBuJajSkjqj0O7SbE8JrpDuoBkbtRKPc8I9nq8AHRsSBiEilMAiNbtLZQADabVoAF1zKoEWVkZj1hBcRFVvgyCSoPZzHiWDMkSJKVjoLTyX9HpQIFDKBIUCjaHTVOZGEA) as a starting point. As long as the types have the a similar structure as in RxJS, it should be reproducible.

**Comment by RomanRolginSky** [comment]
@cartant thank for reply,
I'll try to reproduce the issue with your playground 🙌🏻

**Comment by RomanRolginSky** [comment]
Typescript project issue [link →](https://github.com/microsoft/TypeScript/issues/43962)

**Comment by benlesh** [comment]
Hi @RomanRolginSky — thank you for taking the time to open this issue and for sharing your thoughtful feedback with the RxJS project. I’m sorry it took us so long to respond. Given how much time has passed and the lack of recent activity, I’m going to close this for now so the tracker reflects the work we’re likely to act on. If this is still relevant on a current RxJS release, please feel welcome to open a new issue with updated context and a reproduction. Thank you again for your contribution and patience.

— Ben
